From bf5938a07a5ebba08f51c5465ef18b696692f391 Mon Sep 17 00:00:00 2001
From: Shen Yong <b00984@freescale.com>
Date: Sat, 27 Sep 2008 15:15:09 +0800
Subject: [PATCH] ENGR00093243 fix touch screen driver bug

the kernel thread for ts should not start if adc is not ready

Signed-off-by: Shen Yong<b00984@freescale.com>
---
 drivers/input/touchscreen/mxc_ts.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/input/touchscreen/mxc_ts.c b/drivers/input/touchscreen/mxc_ts.c
index eefd4b9..a654a41 100644
--- a/drivers/input/touchscreen/mxc_ts.c
+++ b/drivers/input/touchscreen/mxc_ts.c
@@ -68,6 +68,9 @@ static int __init mxc_ts_init(void)
 {
 	int retval;
 
+	if (!is_pmic_adc_ready())
+		return -ENODEV;
+
 	mxc_inputdev = input_allocate_device();
 	if (!mxc_inputdev) {
 		printk(KERN_ERR
-- 
1.5.4.4

